From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 29 Oct 2018 01:13:05 +0000
Subject: sk_buff: Avoid ABI change in 4.9.134
Forwarded: not-needed

Commits 316986fe4dca "inet: frags: get rid of ipfrag_skb_cb/FRAG_CB"
and 10043954eada "ip: use rb trees for IP frag queue." moved
sk_buff::sk into a union with a new member, ip_defrag_offset.  This
doesn't change the size or offsets of existing members, so should be
ABI compatible.  Hide the change from genksyms.

---

--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -646,10 +646,14 @@ struct sk_buff {
 		struct rb_node		rbnode; /* used in netem, ip4 defrag, and tcp stack */
 	};
 
+#ifndef __GENKSYMS__
 	union {
 		struct sock		*sk;
 		int			ip_defrag_offset;
 	};
+#else
+	struct sock		*sk;
+#endif
 
 	struct net_device	*dev;
 
